### ISSUE TITLE
defncopy: index names not properly escaped

### ISSUE THREAD

**Issue by jonahgeorge** [FreeTDS/freetds#429]
It appears that `defncopy` does not properly escape index names when writing the create index command.

I was able to reproduce this on an existing table which has a column named `FromDate` and an index on said column named `From Date`:

```sql
CREATE TABLE dbo.tblReportPeriod
	( [RecordID]   int             NOT NULL
	, [FromDate]   nvarchar(40)        NULL
	, [ToDate]     nvarchar(40)        NULL
	)

CREATE  nonclustered INDEX [From Date] on dbo.tblReportPeriod([FromDate])
```

`defncopy` generated the following invalid SQL:
```sql
CREATE TABLE dbo.tblReportPeriod
	( RecordID   int             NOT NULL
	, FromDate   nvarchar(40)        NULL
	, ToDate     nvarchar(40)        NULL
	)

CREATE  nonclustered INDEX From Date on dbo.tblReportPeriod(FromDate)
```

Related
- https://github.com/FreeTDS/freetds/pull/276
- https://github.com/rails-sqlserver/activerecord-sqlserver-adapter/issues/944
- https://github.com/rails-sqlserver/tiny_tds/issues/502

**Comment by jonahgeorge** [comment]
Per comments in the linked PR, if this were to be fixed it appears that `defncopy` would need to know which quoting strategy to use (brackets for MSSQL vs quotes elsewhere). Would a CLI flag be acceptable?

**Comment by fziglio** [comment]
Sybase 15 introduced `[]` quoting so I think it would be safe now to use always `[]` style quoting. One issue of the PR is that it's not properly quoting. If the name is `test]123` it needs to be quoted as `[test]]123]`, not `[test]123]`. There's a `tds_quote_id` to quote identifier (there's also an internal `tds_quote` function).

**Comment by Michoels** [comment]
This is still a problem in 2024.

**Comment by freddy77** [comment]
Fixed in 1.4.12 (will be packaged automatically tomorrow) and master.
